service/test/servicelayerinstance.cc

```diff
@@ -229,9 +229,45 @@ std::vector<chirp::Chirp> ServiceLayerInstance::Monitor(
 }
 std::vector<chirp::Chirp> ServiceLayerInstance::Stream(
     const std::string &hashtag) {
+  /* Get Current TimeStamp */
+  std::time_t seconds;
+  int64_t microseconds_since_epoch;
+  SetTimeStamp(seconds, microseconds_since_epoch);
+
+  auto from_get_function = kvstore->Get("hashtag#" + hashtag);
+  std::vector<chirp::Chirp> chirps_to_send;
+
+  if (from_get_function.size() == 0) {
+    return chirps_to_send;
+  }
+
+  std::set<std::string> chirpsent; /* Save all the id of chirps was sent */
+  chirp::Hashtag tag;
+  tag.ParseFromString(from_get_function[0]);
+
   chirp::StreamReply reply;
 
-  std::set<std::string> chirpsent;
+  /*
+    Continuously look through in db for #hashtag. Keep all
+    sent chirps in a set called chirpsent to disallow sending duplicates. Keep
+    looking for new chirps with this while loop.
+  */
+  int k = 0;
+  while (k < 10) {
```

Comment:
why 10?

Reply:
Oh, I realized why it's 10 because this is the testing class! Disregard my previous question.